internal/xcontext/merge_contexts.go

```diff
@@ -0,0 +1,45 @@
+package xcontext
+
+import (
+	"context"
+	"time"
+)
+
+type MergedContexts struct {
+	context.Context                 //nolint:containedctx
```

Comment:
Do you need context.Context field? I don't see usage of the field.

What about rename:
RequestContext -> DeadlineContext
BaseContext -> AdditionalValues

and make them private

Reply:
done